src/matrix/room/timeline/entries/EventEntry.js

```diff
@@ -24,6 +24,9 @@ export class EventEntry extends BaseEventEntry {
         this._eventEntry = eventEntry;
         this._decryptionError = null;
         this._decryptionResult = null;
+        this._contextEntry = null;
+        this._contextForEntries = null;
```

Comment:
I'm wondering if we can avoid putting a new member variable in this class for a very specific use case (which tends to have a lot of instances, and I'd like to keep light). What about that instead of putting it here, we put the array of entries we are context for into `Timeline._contextEntriesNotInTimeline`, by changing the value from an entry to an entry and an array of entries for which the first entry is the context. Does that make sense?

Reply:
Ah, this won't work because we only put the context entry in `_contextEntriesNotInTimeline` if it wasn't in the remoteEntries already, but in case of an update, we need to know which entries are using the context entry regardless of whether it came from remoteEntries or storage/hs.

Reply:
Ok, never mind, let's keep it as it is :+1: